framework/src/parser/MooseSyntax.C

```diff
@@ -62,12 +63,16 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   /// Variable/AuxVariable Actions
   syntax.registerActionSyntax("AddVariableAction", "Variables/*");
+  syntax.registerSyntaxType("Variables/*", "VariableName");
+  syntax.registerSyntaxType("Variables/*", "NonlinearVariableName");
```

Comment:
We definitely still need `type(name)` as that is what is used for `cpp_type`.
It looks like both `getMooseType(name)` and `getVecMooseType(name)` are used in other places so we can't just remove these.
We can try to clean it up a little but I don't know if this PR is the place to do it since they are seemingly unrelated.